framework/src/iotbus/iotbus_uart.c

```diff
@@ -388,10 +390,24 @@ int iotbus_uart_write(iotbus_uart_context_h hnd, const char *buf, unsigned int l
 	}
 
 	fd = handle->fd;
+
+	tio.loopback = true;
+	ret = loopback(fd, TIOCLOOPBACK, &tio);
+	if (ret < 0) {
```

Comment:
How about adding logs for error cases for easy debugging?
That's because it returns IOTBUS_ERROR_UNKNOWN for failures of loopback, write.
With current implementation, we should put test logs to find failed point whenever this function is failed.

Reply:
i am returning IOTBUS_ERROR_UNKNOWN because we are setting a bit and it will not return any error unless ioctl returns any error and in ioctl we don't know which kind of error it can return.
you can suggest to me what kind of error we can print the logs here?

Reply:
I meant iotbus_uart_write, not ioctl.
If iotbus_uart_write returns IOTBUS_ERROR_UNKNOWN, user can't know where an error occurred because all failures of tcloopback and write return same value.
So I wanted to add some logs. For examples,
ret = tcloopback(fd, &loopback);
if (ret < 0) {
   ibdbg("tcloopback FAILED, %d\n", ret);
  return IOTBUS_ERROR_UNKNOWN;
}
...
ret = write(fd, buf, length);
if (ret < 0) {
  ibdbg("Write ## FAILED, fd %d, errno %d\n", fd, errno);
  return IOTBUS_ERROR_UNKNOWN;
}

Reply:
corrected now